src/NuGetPackageVerifier/Rules/AuthenticodeSigningRule.cs

```diff
@@ -62,21 +58,16 @@ public class AuthenticodeSigningRule : IPackageVerifierRule
             }
             finally
             {
-                CleanUpFolder(nupkgWithoutExt, logger);
+                CleanUpFolder(extractPath, logger);
```

Comment:
You might want to put this in it's own try catch and retry a couple of times. I've had to do this in other tools where I was removing temp directories